SCons/Tool/yacc.py

```diff
@@ -162,6 +162,9 @@ def generate(env):
 
 
 def exists(env):
+    if 'YACC' in env:
```

Comment:
Is it a good idea to include a check that `YACC` is something truthy as well?

Reply:
Or will Detect just deal with that for you?

Reply:
If YACC is garbage Detect will be False (or throw an exception if it's not understandable garbage I suppose)

Reply:
Sounds good. LGTM for sure then. Thanks for the quick fix.